Unstructured-IO/unstructured-api-tools
# chore: adding optional cors to pipeline api

Testing steps!

* activate pyenv for `unstructured-api` (not `-tools`) and cd to root dir of `unstructured-api-tools` to run `pip install -e .` to install the current branch. (should see __version__.py ( 0.10.5 ) from this branch match output from pip list)
* in the `unstructured-api` dir, run make generate-api
* export ALLOWED_ORIGINS="http://example.com/" and start the app locally with make run-web-app
* run curl and POST and OPTION requests to show the Access-Control-Allow-Origin header being set

example curl request:
should see 200:
```
curl -X OPTIONS -H "Origin: http://example.com" -H "Access-Control-Request-Method: POST" -i http://localhost:8000/general/v0/general
```

should see 400 (bad cors origin)
```
curl -X OPTIONS -H "Origin: http://localhost:3000" -H "Access-Control-Request-Method: POST" -i http://localhost:8000/general/v0/general
```
